imagenet/main.py

```diff
@@ -1,3 +1,39 @@
+'''
+Description:
+
+rank: node's rank among all the nodes
+world-size: total node numbers
+start-epoch: don't need pass since training script will load it from checkpoint.
+Pass checkpoint file full path using parameter `--resume`
+
+The program will use all gpus available on the nodes.
+
+Examples to run elastic training:
+
+### Single node, multiple GPUs:
+
+
+```bash
+python main.py -a resnet50 --dist-url 'tcp://127.0.0.1:FREEPORT' --dist-backend 'nccl' --multiprocessing-distributed
+  --world-size 1 --rank 0 --resume=[checkpoint-file-path] --batch-size=32 [imagenet-folder with train and val folders]
+```
+
+### Multiple nodes:
+
+Node 0:
+```bash
+python main.py -a resnet50 --dist-url 'tcp://IP_OF_NODE0:FREEPORT' --dist-backend 'nccl' --multiprocessing-distributed
+  --world-size 2 --rank 0 --resume=[checkpoint-file-path] --batch-size=32  [imagenet-folder with train and val folders]
+```
+
+Node 1:
+```bash
+python main.py -a resnet50 --dist-url 'tcp://IP_OF_NODE0:FREEPORT' --dist-backend 'nccl' --multiprocessing-distributed 
+  --world-size 2 --rank 1 --resume=[checkpoint-file-path] --batch-size=32  [imagenet-folder with train and val folders]
+```
```

Comment:
add an argument like output_path? it will be directory.
we can turn on resume by default.
if there is a ckpt under output_path, resume from it, otherwise, train from scratch.